src/components/course/tests/CourseHeader.test.jsx

```diff
@@ -108,6 +110,40 @@ describe('<CourseHeader />', () => {
     expect(screen.queryByText(shortDescription)).toBeInTheDocument();
   });
 
+  test.each([
+    [['test-skill1', 'test-skill2', 'test-skill3', 'test-skill4'], false],
+    [['test-skill1', 'test-skill2', 'test-skill3', 'test-skill4', 'test-skill5', 'test-skill6'], true],
+  ])('renders course skills', (skills, skillsGreaterThanLimit) => {
+    const courseState = {
+      ...initialCourseState,
+      course: {
+        ...initialCourseState.course,
+        skillNames: skills,
+      },
+    };
+
+    render(
+      <CourseHeaderWithContext
+        initialAppState={initialAppState}
+        initialCourseState={courseState}
+        initialUserSubsidyState={initialUserSubsidyState}
+      />,
+    );
+    skills.forEach((skill, index) => {
+      if (skillsGreaterThanLimit && index >= initialCourseState.maxVisibleSkills) {
```

Comment:
since there are only two cases, I think it is cleaner to have them as two tests, and not have the conditionals in the test. If there are more than a few cases, then yes it makes sense to do this for efficiency and avoid copying

Reply:
Makes sense. I have separated the test cases